packages/lib/services/commands/MenuUtils.ts

```diff
@@ -103,8 +103,16 @@ export default class MenuUtils {
 	}
 
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
-	public commandToStatefulMenuItem(commandName: string, ...args: any[]): MenuItem {
-		const whenClauseContext = this.service.currentWhenClauseContext();
+	public commandToStatefulMenuItem(options: string|CommandToMenuItemOptions, ...args: any[]): MenuItem {
+		let whenClauseContext, commandName;
+		if (typeof options === 'string') {
+			commandName = options;
+			whenClauseContext = this.service.currentWhenClauseContext();
+		} else {
+			commandName = options.commandName;
+			const whenClauseContextOptions = options.whenClauseContextOptions;
+			whenClauseContext = this.service.currentWhenClauseContext(whenClauseContextOptions);
+		}
```

Comment:
I'm not a fan of this. "options" which can be a string, the conditional, etc. It feels messy and it's in a place where code is already quite complex so it would be good to find a cleaner/clearer way to do this